lib/langchain/tool/database.rb

```diff
@@ -0,0 +1,43 @@
+module Langchain::Tool
+  class Database < Base
+    description <<~DESC
+      Useful for getting the result of a database query.
+
+      The input to this tool should be valid SQL.
+    DESC
+
+    # Establish a database connection
+    # example: 'postgres://postgres:postgres@localhost:5432/db_name'
+    # Empty string creates memory database using sqlite3
+    def initialize(db_connection_string = "")
+      depends_on "sequel"
+      depends_on "sqlite3"
+      require "sequel"
+      require "sequel/extensions/schema_dumper"
+
+      @db = if db_connection_string.empty?
+        Sequel.sqlite
+      else
+        Sequel.connect(db_connection_string)
+      end
+      @db.extension :schema_dumper
+    end
+
+    def schema
+      Langchain.logger.info("Database: Getting schema")
+      @db.dump_schema_migration(same_db: true, foreign_keys: false, indexes: false)
+    end
+
+    # Evaluates a sql expression
+    # @param input [String] sql expression
+    # @return [String] results
+    def execute(input:)
+      Langchain.logger.info("Database: Using the Database Tool with \"#{input}\"")
```

Comment:
Would you please modify this to match the format of the following: https://github.com/andreibondarev/langchainrb/blob/main/lib/langchain/tool/serp_api.rb#L36?

Tools == light blue
Agents == red

Reply:
Fixed.